middleware/TestingHandler.md

```diff
@@ -0,0 +1,28 @@
+# Testing Handler
+A piece of middleware that helps to mock different kinds of responses.
+
+## Objectives
+Provide a component, so that application developers can test their application with different kinds of responses (different response codes).
```

Comment:
Why are we limiting it to only response codes? I think we should expand the scope to include response bodies so that we can test the parts of our client that interact with the response body.
@darrelmiller

Reply:
We are returning a response(contains response header and response  body) with a particular response code:
* The responses in case of success scenarios (2xx) are not API specific. So we return an empty response body.
* For other cases we do return the response body containing error parameter.